core/driver/sources/src/main/scala/org/virtuslab/ideprobe/ide/intellij/IntelliJProvider.scala

```diff
@@ -0,0 +1,71 @@
+package org.virtuslab.ideprobe.ide.intellij
+
+import org.virtuslab.ideprobe.IdeProbePaths
+import org.virtuslab.ideprobe.config.{IdeProbeConfig, IntellijConfig}
+import org.virtuslab.ideprobe.dependencies.{IntelliJVersion, InternalPlugins, Plugin}
+
+import java.nio.file.{Files, Path}
+
+trait IntelliJProvider {
+  def factory: IntelliJFactory
+  def plugins: Seq[Plugin]
+  def setup(): InstalledIntelliJ
+  def version: IntelliJVersion
+  def withFactory(factory: IntelliJFactory): IntelliJProvider
+  def withPlugin(plugin: Plugin): IntelliJProvider
+}
+
+case class ExistingIntelliJ(
+    path: Path,
+    plugins: Seq[Plugin],
+    factory: IntelliJFactory
+) extends IntelliJProvider {
+  import org.virtuslab.ideprobe.Extensions._
+  override def setup(): InstalledIntelliJ = {
+    val pluginsDir = path.resolve("plugins")
+    val backupDir = Files.createTempDirectory(path, "plugins")
+    pluginsDir.copyDir(backupDir)
+    factory.installPlugins(InternalPlugins.probePluginForIntelliJ(version) +: plugins, path)
```

Comment:
duplicate logic, surely can be resolved while refactoring the factory.

Reply:
Done.